### ISSUE TITLE
Code lens test button does not pass the correct filter string to `cargo test` 

### ISSUE THREAD

**Issue by aloucks** [rust-analyzer/rust-analyzer#4527]
The name of the trait or struct is omitted from the filter string. 

```rust
pub mod quax {
    pub struct Foo;

    pub trait Bar {
        fn bar(&self);

        /// Bar
        /// ```
        /// # use my_crate::quax::{Foo, Bar};
        ///
        /// let foo = Foo;
        /// foo.baz();
        /// ```
        fn baz(&self) {
            self.bar();
        }
    }

    impl Bar for Foo {
        
        fn bar(&self) {
            println!("bar");
        }
    }
}
```

What happens when pressing the "Play" button for on the doctest:

Filter: `quax::baz`

```text
Executing task: cargo test --doc --package my_crate -- quax::baz --nocapture
...
running 0 tests.
...
```

What I would expect to happen when pressing the "Play" button:

Filter: `quax::Bar::baz`

```text
Executing task: cargo test --doc --package my_crate -- quax::Bar::baz --nocapture
...
running 1 test
test src\lib.rs - quax::Bar::baz (line 1246) ... ok
...
```

**Comment by bnjjj** [comment]
Should be ok in the next release. Fixed by #4491

**Comment by aloucks** [comment]
Confirmed that it's fixed in today's nightly. 👍
